fl4health/utils/partitioners.py

```diff
@@ -43,22 +48,32 @@ def __init__(
         Args:
             number_of_partitions (int): Number of new datasets that we want to break the current dataset into
             unique_labels (List[T]): This is the set of labels through which we'll iterate to perform allocation
-            beta (float): This controls the heterogeneity of the partition allocations. The smaller the beta, the
-                more skewed the label assignments will be to different clients.
             min_label_examples (Optional[int], optional): This is an optional input if you want to ensure a minimum
-                number of labels is present on each partition.
+                number of labels is present on each partition. If prior distribution is provided, this is ignored.
                 NOTE: This does not guarantee feasibility. That is, if you have a very small beta and request a large
                 minimum number here, you are unlikely to satisfy this request. In partitioning, if the minimum isn't
                 satisfied, we resample from the Dirichlet distribution. This is repeated some limited number of times.
                 Otherwise the partitioner "gives up". Defaults to None.
+            beta (Optional[float]): This controls the heterogeneity of the partition allocations. The smaller the beta,
+              the more skewed the label assignments will be to different clients. It is mutually exclusive with given
+              prior distribution..
+            prior_distribution (Optional[Dict[T, np.ndarray]], optional): This is an optional input if you want to
+              provide a prior distribution for the Dirichlet distribution. This is useful if you want to make sure that
+              the partitioning of test data is similar to the partitioning of the training data. Defaults to None. It
+              is mutually exclusive with the beta parameter and min_label_examples.
         """
+        assert beta is not None or prior_distribution is not None, "Either beta or prior must be provided"
+        assert beta is None or prior_distribution is None, "Either beta or prior must be provided, not both"
         self.number_of_partitions = number_of_partitions
         self.unique_labels = unique_labels
         self.n_unique_labels = len(unique_labels)
+        self.prior_distribution = prior_distribution
         self.beta = beta
         self.min_label_examples = min_label_examples if min_label_examples else 0
```

Comment:
We might want to add a warning for the user here that a non-zero `min_label_examples` will be ignored if there is a prior distribution. Something like
```
from logging import WARN
if self.min_label_examples > 0 and self.prior_distribution is not None:
    log(WARN, "A prior distribution has been provided for the partitioner, so min_label_examples will be ignored")
```